### PR TITLE
Fix/unread count mutes

### PR DESCRIPTION
**Pull request by mahboubii**

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -1165,27 +1164,17 @@ export class Channel<
...
-      if (this.getClient().userID === message.user?.id) {
-        continue;
-      }
-      if (m.shadowed) {
-        continue;
-      }
-      if (m.silent) {
-        continue;
-      }
-      if (lastRead == null) {
```
Comment (mahboubii) [inline src/channel.ts:1180]: look like this check never passes but it is logically incorrect, removed.

File: src/client.ts [hunk 2]
```diff
@@ -1684,6 +1692,21 @@ export class StreamChat<
     });
   }
 
+  /** userMuteStatus - check if a user is muted or not, can be used after setUser() is called
+   *
+   * @param {string} targetID
+   * @returns {boolean}
+   */
+  userMuteStatus(targetID: string) {
+    if (!this.user || !this.wsPromise)
```
Comment (mahboubii) [inline src/client.ts:1701]: @vishalnarkhede does this make sense? this function is valid only if you set a user so client.mutes is initialized when the `health.check` is fired.
I think we need the same check for `client._muteStatus()` which handles channel mutes

File: src/client.ts [hunk 3]
```diff
@@ -932,6 +936,10 @@ export class StreamChat<
     if (event.type === 'notification.channel_mutes_updated' && event.me?.channel_mutes) {
       this.mutedChannels = event.me.channel_mutes;
     }
+
+    if (event.type === 'notification.mutes_updated' && event.me?.mutes) {
+      this.mutedUsers = event.me.mutes;
```
Comment (mahboubii) [inline src/client.ts:941]: @ferhatelmas does API always return the whole mutes list or could this be a partial list if the list is long?

**Review by nhannah — APPROVED** [review]
Overall looks good to me, not sure if you want to add that check on _muteStatus you mentioned

**Review by vishalnarkhede — APPROVED** [review]

File: src/channel.ts [hunk 4]
```diff
@@ -1135,27 +1149,12 @@ export class Channel<
...
-        count++;
-        continue;
-      }
-      if (m.created_at > lastRead) {
+    for (let i = 0; i < this.state.messages.length; i += 1) {
```
Comment (ferhatelmas) [inline src/channel.ts:1155]: caching length might help

Comment (ferhatelmas) [reply src/channel.ts:1155]: `i++` looks more idiomatic

File: src/channel.ts [hunk 5]
```diff
@@ -1135,27 +1149,12 @@ export class Channel<
...
+      const message = this.state.messages[i];
+      if (this._countMessageAsUnread(message) && message.created_at > lastRead) {
```
Comment (ferhatelmas) [inline src/channel.ts:1157]: reversing conditions might be faster

File: src/client.ts (hunk 3 above)
Comment (ferhatelmas) [reply src/client.ts:944]: yes, it cuts by 100

File: src/channel.ts [hunk 6]
```diff
@@ -1127,6 +1127,20 @@ export class Channel<
     }
   }
 
+  _countMessageAsUnread(message: {
```
Comment (ferhatelmas) [inline src/channel.ts:1130]: how is channel mute integrated into this calculation ?

File: src/channel.ts (hunk 4 above)
Comment (mahboubii) [reply src/channel.ts:1155]: the performance difference is negligible for caching the length
https://eslint.org/docs/rules/no-plusplus

File: src/channel.ts (hunk 5 above)
Comment (mahboubii) [inline src/channel.ts:1157]:
```suggestion
      if (message.created_at > lastRead && this._countMessageAsUnread(message)) {
```

File: src/channel.ts (hunk 4 above)
Comment (ferhatelmas) [reply src/channel.ts:1155]: makes sense, we can assume this is resolved.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**